src/main/java/com/esri/samples/ogc/browse_wfs_layers/BrowseWfsLayersSample.java

```diff
@@ -0,0 +1,229 @@
+        for (WfsLayerInfo layerInfo : wfsLayerInfos) {
+          wfsLayerNamesListView.getItems().add(layerInfo);
+        }
+
+      } else {
+        Alert alert = new Alert(Alert.AlertType.ERROR, "WFS Service Failed to Load!");
+        alert.show();
+      }
+
+    });
+
+    // populate the list view with layer names
+    wfsLayerNamesListView.setCellFactory(list -> new ListCell<>() {
+
+      @Override
+      protected void updateItem(WfsLayerInfo wfsLayerInfo, boolean bln) {
+        super.updateItem(wfsLayerInfo, bln);
+        if (wfsLayerInfo != null) {
+          String fullNameOfWfsLayer = wfsLayerInfo.getName();
+          String[] split = fullNameOfWfsLayer.split(":");
+          String smallName = split[1];
+          setText(smallName);
+        }
+      }
+    });
+
+    // load the selected layer from the list view when load button is clicked
+    loadButton.setOnAction(e -> {
+      updateMap(wfsLayerNamesListView.getSelectionModel().getSelectedItem());
+
+    });
+
+    // add the list view, button and check box to the control panel
+    controlsVBox.getChildren().addAll(wfsLayerNamesListView, loadButton, checkBox);
+    // add the mapview to the stackpane
+    stackPane.getChildren().addAll(mapView, controlsVBox, progressIndicator);
+    StackPane.setAlignment(controlsVBox, Pos.TOP_LEFT);
+    StackPane.setMargin(controlsVBox, new Insets(10, 0, 0, 10));
+  }
+
+  /**
+   * Adds a WfsLayerInfo to the map's operational layers, with a random color renderer.
+   * @param wfsLayerInfo the WfsLayerInfo that the map will display
+   */
+  private void updateMap(WfsLayerInfo wfsLayerInfo){
+
+    progressIndicator.setVisible(true);
+
+    // clear existing WFSLayerInfos from the map's operational layers
+    map.getOperationalLayers().clear();
+
+    // create a WFSFeatureTable from the WFSLayerInfo
+    WfsFeatureTable wfsFeatureTable = new WfsFeatureTable(wfsLayerInfo);
+
+    // set the feature request mode to manual. The table must be manually populated as panning and zooming won't request features automatically.
+    wfsFeatureTable.setFeatureRequestMode(ServiceFeatureTable.FeatureRequestMode.MANUAL_CACHE);
+    wfsFeatureTable.setAxisOrder(checkBox.isSelected() ? OgcAxisOrder.SWAP : OgcAxisOrder.NO_SWAP);
+
+    // create a feature layer to visualize the WFS features
+    FeatureLayer wfsFeatureLayer = new FeatureLayer(wfsFeatureTable);
+
+    // apply a renderer to the feature layer once the table is loaded (the renderer is based on the table's geometry type)
+    wfsFeatureTable.addDoneLoadingListener(()->{
+      wfsFeatureLayer.setRenderer(getRandomRendererForTable(wfsFeatureTable));
+    });
+
+    // add the layer to the map's operational layers
+    map.getOperationalLayers().add(wfsFeatureLayer);
+
+    // populate the table
+    ListenableFuture<FeatureQueryResult> featureQueryResultListenableFuture = wfsFeatureTable.populateFromServiceAsync(new QueryParameters(), false, null );
```

Comment:
Don't need a variable here since you're not using it. Just chain `addDoneListener` on the end of `populateFromServiceAsync`.